### PR TITLE
Draft: ESP32: Move Certificatons, keys and CD files to other partitions

### PR DESCRIPTION
**Pull request by wqx6**
Currently we place the PAI certification, Device Attestation Certification, DAC keys, and CD in the same nvs partition with the factory configs. We need other partitions to store these files so that the user can replace one file or several files instead of re-generate a new nvs factory-partition binary file. 

In this PR:
* The `DEVICE_ATTESTATION_PARTITION` is added to store the PAI cert, DAC, and DAC public-key.
* The `CERTIFICATION_DECLARATION_PARTITION` is add to store the Certification Declaration.
* The `DAC_PRIVATE_KEY_PARTITION` is added to store the DAC private-key which is flash encrypted.
* The scripts are added to generate the binary files.

### CODE REVIEW

File: src/platform/ESP32/ConfigurationManagerImpl.cpp [hunk 1]
```diff
@@ -122,6 +139,10 @@ CHIP_ERROR ConfigurationManagerImpl::Init()
     SuccessOrExit(MapConfigError(esp_err));
     esp_err = nvs_flash_init_partition(CHIP_DEVICE_CONFIG_CHIP_KVS_NAMESPACE_PARTITION);
     SuccessOrExit(MapConfigError(esp_err));
+    esp_err = nvs_flash_init_partition(CHIP_DEVICE_CONFIG_CHIP_DEVICE_ATTESTATION_PARTITION);
+    SuccessOrExit(MapConfigError(esp_err));
+    esp_err = nvs_flash_init_partition(CHIP_DEVICE_CONFIG_CHIP_CERTIFICATION_DECLARATION_PARTITION);
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/ConfigurationManagerImpl.cpp:144]: CHIP_DEVICE_CONFIG_CHIP_DAC_PRIV_KEY_PARTITION does not need to be inited?  Worth adding a comment saying that and explaining why, so someone doesn't add it here.  Similar for the secure init bits above.

Comment (wqx6) [reply src/platform/ESP32/ConfigurationManagerImpl.cpp:148]: The CHIP_DEVICE_CONFIG_CHIP_DAC_PRIV_KEY_PARTITION is not a nvs sub-type partition in this PR. I use `esp_partition_read` to get the dac private key.

File: src/platform/ESP32/ESP32FactoryDataProvider.cpp [hunk 2]
```diff
@@ -114,18 +128,45 @@ CHIP_ERROR ESP32FactoryDataProvider::GetFirmwareInformation(MutableByteSpan & ou
...
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR ESP32FactoryDataProvider::GetProductAttestationIntermediateCert(MutableByteSpan & outBuffer)
 {
-    size_t certSize;
-    ReturnErrorOnFailure(
-        ESP32Config::ReadConfigValueBin(ESP32Config::kConfigKey_PAICert, outBuffer.data(), outBuffer.size(), certSize));
+    size_t certSize = outBuffer.size();
+    ;
```
[the remark this answers is not in the document]
Comment (wqx6) [reply src/platform/ESP32/ESP32FactoryDataProvider.cpp]: Removed

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/all-clusters-app/esp32/partitions.csv [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # Name,   Type, SubType, Offset,  Size, Flags
 # Note: if you have increased the bootloader size, make sure to update the offsets to avoid overlap
-nvs,      data, nvs,     ,        0x6000,
-otadata,  data, ota,     ,        0x2000,
-phy_init, data, phy,     ,        0x1000,
-ota_0,    app,  ota_0,   ,        1700K,
-ota_1,    app,  ota_1,   ,        1700K,
-ot_storage, data, 0x3a,            , 0x2000,
+dac_prv_key, data, 0xFF,  0xb000,  0x1000, encrypted,
```
Comment (chshu) [inline examples/all-clusters-app/esp32/partitions.csv:3]: Let's keep the example as simple as it is, but enhance the logic to support certs relocation.

**Review by chshu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**